adding dynamic instrumentation faq and add minor clarifications (#15841)
DataDog/documentation: content/en/dynamic_instrumentation/_index.md

is_beta: true
private: true
further_reading:
- link: "/dynamic_instrumentation/faq"
  tag: "Documentation"

>   text: "Learn more about how Dynamic Instrumentation works"

Author: changed tag to FAQ?

- [Unified Service Tagging][5] tags `service`, `env`, and `version` are applied to your deployment.
- Optionally, [Source Code Integration][6] is set up for your service.

> **Note**: `debugger_read` and `debugger_write` permissions required to access Dynamic Instrumentation. While Datadog Administrator have those permissions by default, for more information about default and custom roles and assign roles to users, see [Role Based Access Control][11].

Reviewer: ```suggestion
**Note**: `debugger_read` and `debugger_write` permissions are required to access the Dynamic Instrumentation page. For more information about roles and on how to assign roles to users, see [Role Based Access Control][11].
```
Also, should we rename the roles? 😅

Author: I was thinking the same. but we decided to keep those when we rename the product - will sync with Evgeny to see if that changed.

### Enable Remote Configuration

1. Go to the [Remote Configuration setup page][7] and enable the feature for your organization.
2. Create a key. 
3. Update your `datadog-agent` with the provided configuration snippet.
4. Set `remote_configuration.refresh_interval` to be 5[sec].
{{< tabs >}}
{{% tab "Datadog-agent config.yaml" %}}

> Modify `datadog-agent` configuration:

Reviewer: ```suggestion
Modify `datadog-agent.yaml`:
```
